agent/proxyprocess/daemon_test.go

```diff
@@ -246,6 +255,21 @@ func TestDaemonLaunchesNewProcessGroup(t *testing.T) {
 	// even harder!
 
 	// Let defer clean up the child process(es)
+
+	// Get the NEW child PID
```

Comment:
From what I could tell when debugging the most recent round of orphaned procs, the bulk of them were coming from this missing blob of code in the test.

I was also getting sporadic other orphaned procs, so the rest of the changes try to take care of that with excessive `defer destroy()` in the happy-path and a built-in 2 minute self destruct in the not-happy-path.